yfquotes@thegli/files/yfquotes@thegli/desklet.js

```diff
@@ -0,0 +1,261 @@
+/*
+ * Yahoo Finance quotes - 0.0.1, May 2018
+ *
+ * Shows stock quotes information provided by Yahoo Finance.
+ * This desklet is based on the work of fthuin's stock desklet.
+ * 
+ */
+/* jshint esversion : 6  */
+const Desklet = imports.ui.desklet; // cinnamon desklet user interface
+const St = imports.gi.St; // Shell toolkit library from GNOME
+const Gio = imports.gi.Gio; // URL-IO-Operations
+const GLib = imports.gi.GLib; // Files operations
+const Gtk = imports.gi.Gtk; // Gtk library (policies for scrollview)
+const Mainloop = imports.mainloop; // For repeated updating
+const Lang = imports.lang; // Binding desklet to mainloop function
+const Settings = imports.ui.settings; // Settings loader based on settings-schema.json file
+
+var console = global; // So we can use console.log
+var dirPath = "yfquotes@thegli";
+var deskletDir = GLib.get_home_dir() + "/.local/share/cinnamon/desklets/" + dirPath;
+
+var YahooQueryStockQuoteReader = function () {
+};
+
+YahooQueryStockQuoteReader.prototype = {
+	constructor: YahooQueryStockQuoteReader,
+	yahooQueryBaseUrl: "https://query1.finance.yahoo.com/v7/finance/quote?symbols=",
+	getStockQuotes(quoteSymbols) {
```

Comment:
ES6 method definitions only work 100% in Cinnamon 3.8, based on mozjs52. mozjs38 also has this, but some people (usually Debian users) run Cinnamon 3.4-3.6 with CJS 3.2, based on mozjs24. You can specify compatibility in by adding `"cinnamon-version": ["3.8"]` to metadata.json.

Reply:
Thank you for the feedback.
I can revert the method definition changes. Would it then run properly with CJS 3.2 and mozjs-24? I have currently no such setup at hand to test myself.
And does the compatibility flag mean "works up to that Cinnamon version"?

Reply:
No problem - I would have to look more closely. Setting `cinnamon-version` that way would restrict it to being installed on Cinnamon 3.8. There's also another option you can use, `js-version`, you could also try adding `"js-version": ["3.4.0", "3.6.0", "3.8.0"]` to metadata.json. No xlets use it AFAIK and I haven't tested it, so can't vouch that it works reliably. If you do go that route I can test in VM to make sure its working.